pkg/controller/appsync/appsync.go

```diff
@@ -43,9 +49,95 @@ func New(config *config.AppSyncConfig, db *database.Database, h *render.Renderer
 }
 
 // HandleSync performs the logic to sync mobile apps.
-func (c *Controller) HandleSync(ctx context.Context) http.Handler {
+func (c *Controller) HandleSync() http.Handler {
+	type AppSyncResult struct {
+		OK     bool    `json:"ok"`
+		Errors []error `json:"errors,omitempty"`
+	}
+
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// TODO(whaught): implement this
-		controller.InternalError(w, r, c.h, errors.New("not implemented"))
+		logger := logging.FromContext(r.Context()).Named("appsync.HandleSync")
+
+		if c.config.AppSyncURL == "" {
+			controller.InternalError(w, r, c.h, errors.New("no APP_SYNC_URL configured"))
+			return
+		}
+
+		client := http.Client{Timeout: c.config.Timeout}
+		resp, err := client.Get(c.config.AppSyncURL)
+		if err != nil {
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+
+		var apps AppsResponse
+		defer resp.Body.Close()
+
+		if err := json.NewDecoder(io.LimitReader(resp.Body, c.config.FileSizeLimitBytes)).Decode(&apps); err != nil {
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+
+		var merr *multierror.Error
+		realms := map[string]*database.Realm{}
+		appsByRealm := map[uint][]*database.MobileApp{}
+
+		for _, app := range apps.Apps {
```

Comment:
and extract this to a function to make it testable

Reply:
I even added a test!